Update Contributor Guidelines (#82)
Gilead-BioStats/gsm.core: .github/CONTRIBUTING.md

# Introduction

This page outlines the development process for `{gsm}` packages. It summarizes both our project management and development workflows. The project management workflow focuses on using issues to are capture user requirements, bugs and technical requirements. The development workflow explains how to submit code using the GitHub flow paradigm.

# Project Management

The project management workflow focuses on using issues to capture user requirements, bugs, and technical requirements. We also use the [gsm Roadmap](https://github.com/orgs/Gilead-BioStats/projects/41) GitHub Project to track issues and set priorities across repositories. 

## Issues

Issues are the primary way to communicate what needs to be done and to track progress. Several issue templates are provided to help streamline this process, including: 

- Requirements: Use this template to create a User Requirement. Requirements are then assigned sub-issues using the issue types below. 
- Bug: Bug reports for when something isn't working
- Feature: User-facing Functionality
- Technical Task: Non-user facing tasks, such as infrastructure updates or internal tooling improvements.

> - Documentation Task: Improvements or additions to the documentation including function docs, readme updates and vignettes.

Reviewer: ```suggestion
- Requirements – Use this template to create a User Requirement. Requirements are then assigned sub-issues using the issue types below. 
- Bug – Bug reports for when something isn't working
- Feature – User-facing Functionality
- Technical Task – Non-user facing tasks, such as infrastructure updates or internal tooling improvements.
- Documentation Task – Improvements or additions to the documentation including function docs, readme updates and vignettes.
```
One more bit of consistency.

8. **grail** – Suggests `gsm.core`, `gsm.mapping`, `gsm.kri` and `gsm.reporting`.
9. **grail.ado** - Depends on `grail` for passing data to ADO.
10. **gsm.rrm** - Suggests `grail` and `gsm.core`.
11. **gsm.template** – Depends on `gsm.core`, `gsm.datasim` and `gsm.mapping`.
12. **gsm.app** – Depends on `gsm.core` and `gsm.kri`. Suggests `gsm.mapping` and `gsm.reporting`.`
13. **gsm.ae** – Depends on `gsm.app` and `gsm.mapping`
14. **gsm.qc** – Depends on `gsm.core`, `gsm.mapping`, `gsm.kri` and `gsm.reporting`.
15. **gsm.utils** – Optional utilities. No gsm dependencies at this time

**Note:** Not every package requires a release in every cycle. If no changes have been made (and the version number/NEWS.md do not need updating), simply skip that package and continue with the next in the sequence.

By following this order, you ensure that every package is released with its latest compatible dependencies already available, minimizing risk of conflicts and downstream errors.

---
  
## Post-Release Steps
  
* Verify that all release tags are visible on GitHub.
* Add qualification documentation to `r-qualification` repo, as required.

> * Update [Package Tracker in Analytics WG spreadsheet](https://gileadconnect.sharepoint.com/:x:/r/teams/RiskBasedMonitoringInitiative/Shared%20Documents/Analytics_Technology/Analytics%20Working%20Groups/Analytics%20WG%20Tracker.xlsx?d=w8bc03988c6764d66ba2b769648a0f496&csf=1&web=1&e=Qlc7Nw&nav=MTVfezI1RDA4QURDLUY0ODItNEE3Mi05QTZCLTBCMzY5QjdERTYzQ30) with release dates, links to releases and qualification information.

Reviewer: Hmm, this is useful for us, but won't work for anybody other than us.

Reviewer 2: ah true...hopefully this spreadsheet is going away soon anyway, so i can remove it and maybe just add it to some internal documentation

Reviewer 2: You okay with removing this from the contributor guidelines, @nandriychuk?

Reviewer 2: I can also make it more generic, like 
```suggestion
* Update the Package Release Tracker with release dates, links to releases and qualification information. *Gilead Only*
```

Reviewer 3: Yes, I'm totally fine with removing the link it so that we don't have to update these guidelines once the spreadheet is no longer used.
